json-rpc/spec.json

```diff
@@ -196,6 +196,101 @@
 				"$ref": "#/components/contentDescriptors/GasPrice"
 			}
 		},
+		{
+			"name": "eth_feeHistory",
+			"summary": "Transaction fee history",
+			"description": "Returns transaction base fee and effective priority fee history for the requested/supported block range.",
+			"params": [
+				{
+					"name": "blockCount",
+					"description": "Number of blocks in the requested range.",
+					"required": true,
+					"schema": {
+						"$ref": "#/components/schemas/Integer"
+					}
+				},
+				{
+					"name": "lastBlock",
+					"description": "Last block number of the requested range.",
+					"required": true,
+					"schema": {
+						"oneOf": [
+							{
+								"$ref": "#/components/schemas/BlockNumber"
+							},
+							{
+								"$ref": "#/components/schemas/BlockNumberTag"
+							}
+						]
+					}
+				},
+				{
+					"name": "rewardPercentiles",
+					"description": "A list of percentile values (0-100) to sample from each block's miner rewards in ascending order, weighted by gas used.",
+					"schema": {
+						"title": "rewardPercentiles",
+						"type": "array",
+						"items": {
+							"type": "number"
+						}
+					}
+				}
+			],
+			"result": {
+				"name": "feeHistoryResult",
+				"description": "Fee history for the returned block range (the intersection of requested and supported range).",
+				"schema": {
+					"title": "feeHistoryResults",
+					"description": "Fee history results.",
+					"type": "object",
+					"required": [
+						"firstBlock",
+						"baseFee",
+						"gasUsedRatio"
+					],
+					"properties": {
+						"firstBlock": {
+							"title": "firstBlock",
+							"description": "First block number of the returned range.",
+							"$ref": "#/components/schemas/BlockNumber"
+						},
+						"baseFee": {
+							"title": "baseFeeArray",
+							"description": "An array of block base fees (including the next block after the returned range).",
```

Comment:
It seems odd to me that this would return *more* blocks than requested.  If the user wants +1, they can just ask for +1.

Reply:
This info is derived from the requested blocks (the next base fee always follows from the last block). It makes sense to specify it like this because the pending base fee is basically the most important and it has nothing to do with the pending state that is not supported at all by some backends (so it either couldn't be queried or would be an ugly corner case with only the baseFee returned). On the other hand if pending state is supported and the pending block is queried then we can even predict even the after-pending block's base fee because it follows from the pending state and it is a good indicator of congestion in the mempool. There isn't even a way to specify the after-pending block as lastBlock. So I'd strongly like to keep it this way and specify that the function returns info derivable from the requested blocks (hence the extra baseFee value).